examples/src/main/java/com/datastax/oss/driver/examples/paging/RandomPagingRestUi.java

```diff
@@ -0,0 +1,378 @@
+   *
+   * <p>Note that in this example, we rely on the automatic discovery and configuration of Jackson
+   * through {@code org.glassfish.jersey.jackson.JacksonFeature}.
+   */
+  public static class VideoApplication extends ResourceConfig {
+
+    public VideoApplication(final CqlSession session) {
+      super(UserService.class);
+      // AbstractBinder is provided by HK2
+      register(
+          new AbstractBinder() {
+
+            @Override
+            protected void configure() {
+              bind(session).to(CqlSession.class);
+            }
+          });
+    }
+  }
+
+  /**
+   * A typical REST service, handling requests involving users.
+   *
+   * <p>Typically, this service would contain methods for listing and searching for users, and
+   * methods to retrieve user details. Here, for brevity, only one method, listing videos by user,
+   * is implemented.
+   */
+  @Singleton
+  @Path("/users")
+  @Produces("application/json")
+  public static class UserService {
+
+    @Inject private CqlSession session;
+
+    @Context private UriInfo uri;
+
+    private PreparedStatement videosByUser;
+    private Pager pager;
+
+    @PostConstruct
+    @SuppressWarnings("unused")
+    public void init() {
+      this.pager = new Pager(session, ITEMS_PER_PAGE);
+      this.videosByUser =
+          session.prepare(
+              "SELECT videoid, title, added FROM examples.random_paging_rest_ui WHERE userid = ?");
+    }
+
+    /**
+     * Returns a paginated list of all the videos created by the given user.
+     *
+     * @param userid the user ID.
+     * @param page the page to request, or {@code null} to get the first page.
+     */
+    @GET
+    @Path("/{userid}/videos")
+    public UserVideosResponse getUserVideos(
+        @PathParam("userid") int userid, @QueryParam("page") Integer page) {
+
+      Statement statement = videosByUser.bind(userid).setPageSize(FETCH_SIZE);
+
+      if (page == null) {
+        page = 1;
+      }
+      ResultSet rs = pager.skipTo(statement, page);
+
+      List<UserVideo> videos;
+      boolean empty = isExhausted(rs);
+      if (empty) {
+        videos = Collections.emptyList();
+      } else {
+        int remaining = ITEMS_PER_PAGE;
+        videos = new ArrayList<>(remaining);
+        for (Row row : rs) {
+          UserVideo video =
+              new UserVideo(row.getInt("videoid"), row.getString("title"), row.getInstant("added"));
+          videos.add(video);
+
+          if (--remaining == 0) {
+            break;
+          }
+        }
+      }
+
+      URI previous =
+          (page == 1) ? null : uri.getAbsolutePathBuilder().queryParam("page", page - 1).build();
+      URI next = (empty) ? null : uri.getAbsolutePathBuilder().queryParam("page", page + 1).build();
+      return new UserVideosResponse(videos, previous, next);
+    }
+  }
+
+  private static boolean isExhausted(ResultSet rs) {
+    return !rs.iterator().hasNext();
+  }
```

Comment:
Could you inline this? Feels a bit random to have it extracted in a static method, it's only used in one place.